hc_dpki/src/keypair.rs

```diff
@@ -1,610 +1,242 @@
-    ///
-    /// @param {string} hint - additional info / description for the bundle
-    pub fn get_bundle(
-        &mut self,
-        passphrase: &mut SecBuf,
-        hint: String,
-        config: Option<PwHashConfig>,
-    ) -> Result<bundle::KeyBundle, HolochainError> {
-        let bundle_type: String = "hcKeypair".to_string();
-        let corrected_pub_keys = KeyBuffer::with_corrected(&self.pub_keys)?;
+    fn public(&self) -> Base32;
+    fn private(&mut self) -> &mut SecBuf;
+    fn new_from_seed(seed: &mut SecBuf) -> HcResult<Self>
+    where
+        Self: Sized;
+    fn codec() -> HcidEncoding;
 
-        let mut key_buf = SecBuf::with_secure(BUNDLE_DATA_LEN);
+    // -- Common methods -- //
 
-        let mut offset: usize = 0;
+    /// Decode the public key from Base32 into a [u8]
+    fn encode_pub_key(pub_key_sec: &mut SecBuf) -> Base32 {
+        utils::encode_pub_key(pub_key_sec, &Self::codec()).expect("Public key encoding failed.")
+    }
 
-        key_buf.write(offset, &[1])?;
-        offset += 1;
+    /// Decode the public key from Base32 into a [u8]
+    fn decode_pub_key(&self) -> Vec<u8> {
+        Self::codec()
+            .decode(&self.public())
+            .expect("Public key decoding failed. Key was not properly encoded.")
+    }
 
-        key_buf.write(1, corrected_pub_keys.get_sig())?;
-        offset += sign::PUBLICKEYBYTES;
+    /// Decode the public key from Base32 into a SecBuf
+    fn decode_pub_key_into_secbuf(&self) -> SecBuf {
+        utils::decode_pub_key(self.public(), &Self::codec())
+            .expect("Public key decoding failed. Key was not properly encoded.")
+    }
 
-        key_buf.write(offset, corrected_pub_keys.get_enc())?;
-        offset += kx::PUBLICKEYBYTES;
+    /// Return true if the keys are equivalent
+    fn is_same(&mut self, other: &mut Self) -> bool {
+        self.public() == other.public() && self.private().is_same(other.private())
+    }
+}
 
-        key_buf.write(offset, &**self.sign_priv.read_lock())?;
-        offset += sign::SECRETKEYBYTES;
+//--------------------------------------------------------------------------------------------------
+// Signing KeyPair
+//--------------------------------------------------------------------------------------------------
 
-        key_buf.write(offset, &**self.enc_priv.read_lock())?;
+/// KeyPair used for signing data
+pub struct SigningKeyPair {
+    pub public: Base32,
+    pub private: SecBuf,
+}
 
-        let password_encrypted: bundle::ReturnBundleData =
-            util::pw_enc(&mut key_buf, passphrase, config)?;
-        let bundle_data_serialized = json::encode(&password_encrypted).unwrap();
+impl KeyPair for SigningKeyPair {
+    fn public(&self) -> String {
+        self.public.clone()
+    }
+    fn private(&mut self) -> &mut SecBuf {
+        &mut self.private
+    }
+    fn codec() -> HcidEncoding {
+        with_hcs0().expect("HCID failed miserably with_hcs0.")
```

Comment:
lazy_static

Reply:
Ok, but shouldn't HCID crate provide those?

Reply:
indeed, that might make the most sense. I have some vague unfounded worry that lazy_static and wasm compilation don't work well together... I wonder where that came from... but it would totally be worth trying.